cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -59,3 +67,28 @@ func Run(gadgetRoot, device string, options *Options) error {
 
 	return nil
 }
+
+func ensureLayoutCompatibility(gadgetLayout, diskLayout *gadget.LaidOutVolume) error {
+	eq := func(ds, gs gadget.LaidOutStructure) bool {
+		dv := ds.VolumeStructure
+		gv := gs.VolumeStructure
+		return dv.Name == gv.Name && dv.Role == gv.Role && ds.StartOffset == gs.StartOffset && dv.Size == gv.Size && dv.Filesystem == gv.Filesystem
```

Comment:
Should we check the filesystem label as well?

Where does the role information come from in the disk layout?

Reply:
We can't also check the label because certain structures are now unlabeled in the gadget (system-seed and system-data). The role information should indeed not be checked because it's not retrieved from the device, however the spread test works because the gadget used there doesn't specify structure roles. Nice catch! This test must be updated once we have a core20-compiliant pc gadget available.